### DISCUSSION TITLE
proxySet `set.delete()` can not trigger re-render

### DISCUSSION THREAD

**Discussion by magicdawn** [pmndrs/valtio#979] [Bug report]
### Bug Description

![01](https://github.com/user-attachments/assets/08065304-142d-41a5-9381-b2b3cca78ddd)

## steps
- initial: shows `included = false`
- after click add, showing `included = true`, as expected
- after click remove, showing `included = true`, expected `included = false`

### Reproduction Link

https://codesandbox.io/p/sandbox/4tcs96

**Comment by dai-shi** [comment]
@overthemike would you like to take a look?

**Reply by overthemike** [reply]
Yeah I'll take a look.

**Comment by magicdawn** [comment]
more info
- `.delete` / `.clear` did not update UI
- this demo works with valtio@2.0.0, does not work with valtio@2.1.0

**Comment by overthemike** [comment]
@magicdawn thanks for the report. This was actually an issue in both proxyMap and proxySet. The issue was in the `has` method.
#981

**Comment by magicdawn** [comment]
I found access `.size` can "eliminate" this issue

![02](https://github.com/user-attachments/assets/5b18f982-d014-4a1d-a8b4-a32f69cca1e6)

and I have this demo as unit test in https://github.com/magicdawn/valtio/commit/95ca1b5646035130e553444fcdafdac1120202ff 
this test does fail on main branch, would u mind pick this test up

**Reply by overthemike** [reply]
This was actually fixed in #981 and shouldn't need the size hack you showed above. #988 was a similar fix for iterator methods.
